doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -668,69 +652,83 @@ Induction
       implicitly added to this table when defined.
       Use the :cmd:`Add` and :cmd:`Remove` commands to update this set manually.
 
-.. tacn:: inversion @ident
-   :name: inversion
+:tacn:`inversion`
+~~~~~~~~~~~~~~~~~
+
+.. tacn:: inversion {| @ident | @natural } using @one_term {? in {+ @ident } }
+          inversion {| @ident | @natural } {? as @or_and_intropattern } {? in {+ @ident } }
+   :name: inversion; _
+
+   .. comment: the other inversion* tactics don't support the using clause,
+      but they should be able to, if desired.  It wouldn't make sense for
+      inversion_sigma.
+      See https://github.com/coq/coq/pull/14179#discussion_r642193096
+
+   .. todo PR: following is clear as mud
 
    Let the type of :n:`@ident` in the local context be :g:`(I t)`, where :g:`I`
-   is a (co)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
+   is a (co-)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
```

Comment:
That doesn't look right to me, are we writing "co-inductive" anywhere else? I believe it should always be coinductive.

Reply:
Currently in the doc:
coinductive: 12 uses
co-inductive: 38 uses
Googling suggests that "coinductive" is preferred.  I will use that.

FWIW, I just ran across this 60 page [A Tutorial on [Co-]Inductive Types in Coq](https://www.labri.fr/perso/casteran/RecTutorial.pdf).  If it's not totally out of date
(it refers to Coq 8.0), perhaps worth including a link to it somewhere in the doc?

Reply:
Did the same for `co-recursion` and `co-fixpoint`.  Not so sure about `co-pattern`:

> "Such a technique is akin to the *co-pattern* style that can be
found in e.g. Agda, and preserves subject reduction."

Reply:
I think copattern is more correct and FWIW Agda agrees too: https://agda.readthedocs.io/en/v2.6.0.1/language/copatterns.html

Reply:
> FWIW, I just ran across this 60 page [A Tutorial on [Co-]Inductive Types in Coq](https://www.labri.fr/perso/casteran/RecTutorial.pdf). If it's not totally out of date (it refers to Coq 8.0), perhaps worth including a link to it somewhere in the doc?

I don't know how up-to-date / out-of-date this is (probably worth discussing in another issue cc-ing its author `@Casteran` and `@palmskog`) but I know that this associated Coq file is being maintained: https://github.com/coq-community/coq-art/blob/master/tutorial_inductive_co_inductive_types/SRC/RecTutorial.v
That being said, there might be other reasons to consider this tutorial out-of-date, e.g., if it doesn't mention the most frequently used techniques for managing coinductive types (e.g. paco?).